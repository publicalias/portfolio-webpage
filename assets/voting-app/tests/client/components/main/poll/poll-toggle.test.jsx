"use strict";

//local imports

const PollToggle = require("../../../../../scripts/client/components/main/poll/poll-toggle");

const { initTestPoll, testReload, testWrapper } = require("../../../test-helpers");

//global imports

const { newUser } = require("schemas");
const { initTestSnapshot, reactTests } = require("test-helpers/react-tests");

//utilities

const { testMount, testShallow } = testWrapper(PollToggle);

const testSnapshot = initTestSnapshot(testShallow);

const initTestProp = (render) => (prop) => render({ user: newUser({ id: "id-a" }) }, prop && {
  users: {
    [prop]: ["id-a"]
  }
});

const initTestToggle = (render) => (qa, type, list) => {

  const dataList = [null, { id: "id-a" }, list];

  return testReload(render, dataList, qa, list ? null : "id-a", [type, ["id-a"]]);

};

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(PollToggle));

//poll toggle

describe("poll toggle (flag)", () => {

  const testFlag = initTestPoll(testSnapshot, "flag");
  const testFlagMount = initTestPoll(testMount, "flag");

  const testProp = initTestProp(testFlag);
  const testToggle = initTestToggle(testFlagMount);

  it("should match snapshot (default)", () => testProp());

  it("should match snapshot (toggled)", () => testProp("flagged"));

  it("should call pollToggleFlag on click (list)", () => testToggle(".qa-toggle-flag", "pollToggleFlag", true));

  it("should call pollToggleFlag on click (view)", () => testToggle(".qa-toggle-flag", "pollToggleFlag"));

});

describe("poll toggle (hide)", () => {

  const testHide = initTestPoll(testSnapshot, "hide");
  const testHideMount = initTestPoll(testMount, "hide");

  const testProp = initTestProp(testHide);
  const testToggle = initTestToggle(testHideMount);

  it("should match snapshot (default)", () => testProp());

  it("should match snapshot (toggled)", () => testProp("hidden"));

  it("should call pollToggleHide on click (list)", () => testToggle(".qa-toggle-hide", "pollToggleHide", true));

  it("should call pollToggleHide on click (view)", () => testToggle(".qa-toggle-hide", "pollToggleHide"));

});
