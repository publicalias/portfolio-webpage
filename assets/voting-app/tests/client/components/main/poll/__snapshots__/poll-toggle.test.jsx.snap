// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`poll toggle (flag) should match snapshot (default) 1`] = `
<button
  className="c-icon-button qa-toggle-flag "
  onClick={[Function]}
>
  <i
    className="far fa-flag"
  />
</button>
`;

exports[`poll toggle (flag) should match snapshot (toggled) 1`] = `
<button
  className="c-icon-button qa-toggle-flag "
  onClick={[Function]}
>
  <i
    className="fas fa-flag"
  />
</button>
`;

exports[`poll toggle (flag) should match snapshot (util) 1`] = `
<button
  className="c-icon-button qa-toggle-flag u-whatever"
  onClick={[Function]}
>
  <i
    className="far fa-flag"
  />
</button>
`;

exports[`poll toggle (hide) should match snapshot (default) 1`] = `
<button
  className="c-icon-button qa-toggle-hide "
  onClick={[Function]}
>
  <i
    className="far fa-eye"
  />
</button>
`;

exports[`poll toggle (hide) should match snapshot (toggled) 1`] = `
<button
  className="c-icon-button qa-toggle-hide "
  onClick={[Function]}
>
  <i
    className="far fa-eye-slash"
  />
</button>
`;

exports[`poll toggle (hide) should match snapshot (util) 1`] = `
<button
  className="c-icon-button qa-toggle-hide u-whatever"
  onClick={[Function]}
>
  <i
    className="far fa-eye"
  />
</button>
`;
